src/core/crypto/crypto_platform.cpp

```diff
@@ -49,6 +49,13 @@
 using namespace ot;
 using namespace Crypto;
 
+static mbedtls_aes_context sAesContext;
+#if !OPENTHREAD_RADIO
+static mbedtls_md_context_t   sHmacSha256Context;
+static HmacSha256::Hash       sHkdfContext;
+static mbedtls_sha256_context sSha256Context;
```

Comment:
Thanks @gjc13. I like the idea behind this optimization, but I think there may some limitations thet we need to consider.

Here we are sharing one context for all operations whereas before this PR the context is tracked by the each crypto object. So for example, with the current PR we cannot have multiple SHA-256 calculations happen in parallel. 

I am not sure if we currently use parallel SHA calculations (or other crypto API) in the code but we certainly don't restrict such a use with our current `Crypto::Sha256` class definition (any module can have its own `Crypto::Sha256` instance and use it to perform calculations).